apps/public/src/app/(general)/bijlage/Attachments.tsx

```diff
@@ -0,0 +1,131 @@
+'use client'
+
+import {
+  Column,
+  ErrorMessage,
+  Field,
+  FileList,
+  Heading,
+  Label,
+  Paragraph,
+  UnorderedList,
+} from '@amsterdam/design-system-react'
+import type { ApiError } from '@meldingen/api-client'
+import { Grid, SubmitButton } from '@meldingen/ui'
+import { useRef, useState } from 'react'
+import type { ChangeEvent } from 'react'
+
+import {
+  deleteMeldingByMeldingIdAttachmentByAttachmentId,
+  postMeldingByMeldingIdAttachment,
+} from 'apps/public/src/apiClientProxy'
+
+import { BackLink } from '../_components/BackLink'
+
+import { FileInput } from './_components/FileInput'
+import { redirectToNextPage } from './actions'
+import styles from './Attachments.module.css'
+
+const MAX_FILES = 3
+
+type Props = {
+  meldingId: number
+  token: string
+}
+
+export type UploadedFiles = { file: File; id: number }
+
+export const Attachments = ({ meldingId, token }: Props) => {
+  const formRef = useRef<HTMLFormElement>(null)
+  const [uploadedFiles, setUploadedFiles] = useState<UploadedFiles[]>([])
+  const [errorMessage, setErrorMessage] = useState<string>()
+
+  const handleOnChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage(undefined)
+
+    if (!event.currentTarget.files) return
+
+    const files = Array.from(event.currentTarget.files)
+
+    try {
+      if (files.length + uploadedFiles.length > MAX_FILES) {
+        throw new Error(`U kunt maximaal ${MAX_FILES} bestanden uploaden.`)
+      }
+
+      const result = await Promise.all(
+        files.map(async (file) => {
+          const uploadedFile = await postMeldingByMeldingIdAttachment({
+            formData: { file },
+            meldingId,
+            token,
+          })
+
+          return { file, id: uploadedFile.id }
+        }),
+      )
+      setUploadedFiles((currentFiles) => [...currentFiles, ...result])
+    } catch (error) {
+      formRef.current?.reset()
```

Comment:
This should be outside of the `try...catch` block I think. Resetting the form should also happen on success.